x-pack/platform/packages/shared/kbn-elastic-assistant-common/impl/schemas/knowledge_base/entries/common_attributes.gen.ts

```diff
@@ -56,7 +56,7 @@ export const Metadata = z.object({
   /**
    * Knowledge Base resource name for grouping entries, e.g. 'esql', 'lens-docs', etc
    */
-  kbResource: z.string(),
+  kbResource: z.union([z.literal('user'), z.literal('esql')]),
```

Comment:
I think you will need to update it here instead https://github.com/elastic/kibana/blob/main/x-pack/platform/packages/shared/kbn-elastic-assistant-common/impl/schemas/knowledge_base/entries/common_attributes.schema.yaml#L52